frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeaderButtonRow/DashboardHeaderButtonRow.tsx

```diff
@@ -0,0 +1,85 @@
+import { dashboardActionButtons } from "metabase/dashboard/components/DashboardHeader/DashboardHeaderButtonRow/action-buttons";
+import type {
+  DashboardActionKey,
+  DashboardHeaderButtonRowProps,
+  HeaderButtonProps,
+} from "metabase/dashboard/components/DashboardHeader/DashboardHeaderButtonRow/types";
+import {
+  getDashboardComplete,
+  getHasModelActionsEnabled,
+  getIsEditing,
+} from "metabase/dashboard/selectors";
+import { useSelector } from "metabase/lib/redux";
+import { getPulseFormInput } from "metabase/pulse/selectors";
+import {
+  canManageSubscriptions as canManageSubscriptionsSelector,
+  getUserIsAdmin,
+} from "metabase/selectors/user";
+import { Box } from "metabase/ui";
+
+import { DASHBOARD_VIEW_ACTIONS, DASHBOARD_EDITING_ACTIONS } from "./constants";
+
+export const DashboardHeaderButtonRow = ({
+  dashboardActionKeys = null,
+  isPublic = false,
+  isAnalyticsDashboard = false,
+  ...props
+}: {
+  dashboardActionKeys?: DashboardActionKey[] | null;
+} & DashboardHeaderButtonRowProps) => {
+  const formInput = useSelector(getPulseFormInput);
+  const isAdmin = useSelector(getUserIsAdmin);
+  const canManageSubscriptions = useSelector(canManageSubscriptionsSelector);
+
+  const dashboard = useSelector(getDashboardComplete);
+  const canEdit = Boolean(dashboard?.can_write && !dashboard?.archived);
+
+  const hasModelActionsEnabled = useSelector(getHasModelActionsEnabled);
+
+  const isEditing = useSelector(getIsEditing);
+
+  const buttonOptions = isEditing
+    ? DASHBOARD_EDITING_ACTIONS
+    : DASHBOARD_VIEW_ACTIONS;
+
+  const visibleKeys = dashboardActionKeys
+    ? buttonOptions.filter(key => dashboardActionKeys.includes(key))
+    : buttonOptions;
+
+  return (
+    <>
+      {visibleKeys.map(key => {
```

Comment:
I think `key` sounds too generic, you already named the prop `dashboardActionKeys`, I think this could be improved to
```suggestion
      {visibleActionKeys.map(actionKey => {
```

or the full name, despite a bit long
```suggestion
      {visibleDashboardActionKeys.map(dashboardActionKey => {
```

Reply:
explicit is always better than implicit so let's go with the second option